src/matrix/compressed-matrix.cc

```diff
@@ -37,6 +37,17 @@ MatrixIndexT CompressedMatrix::DataSize(const GlobalHeader &header) {
   }
 }
 
+// scale all element of matrix by scaling floats
+// in GlobalHeader with alpha.
+void CompressedMatrix::Scale(float alpha) {
```

Comment:
It's OK about the CompresssedMatrix::Scale().  It's about the only thing that you can easily do to the compressed matrix without having to uncompress and then compress it.